### PR TITLE
teamWaitingKey refactor to use teamID

### PR DESCRIPTION
**Pull request by taruti**

### CODE REVIEW

File: shared/actions/teams.tsx [hunk 1]
```diff
@@ -178,7 +179,7 @@ const setWelcomeMessage = async (action: TeamsGen.SetWelcomeMessagePayload, logg
 }
 
 const getTeamRetentionPolicy = async (
-  state: TypedState,
+  _: TypedState,
```
Comment (chrisnojima) [inline shared/actions/teams.tsx]: fyi if these don't need state you can just register it with `chainAction` and state won't be passed as the first param

File: shared/actions/teams.tsx [hunk 2]
```diff
@@ -64,8 +64,9 @@ const showTeamAfterCreation = (action: TeamsGen.TeamCreatedPayload) => {
   ]
 }
 
-function* joinTeam(_: TypedState, action: TeamsGen.JoinTeamPayload) {
+function* joinTeam(state: TypedState, action: TeamsGen.JoinTeamPayload) {
   const {teamname} = action.payload
+  const teamID = Constants.getTeamID(state, teamname)
```
Comment (buoyad) [inline shared/actions/teams.tsx]: This most likely won't work, we won't have the teamID for this team in the store. I couldn't find anything using the waiting key for this action though, so I think you could just remove it.

File: shared/actions/teams.tsx (hunk 1 above)
Comment (taruti) [reply shared/actions/teams.tsx]: Refactoring those to use `s/chainAction2/chainAction/`. Is there a good solution to `chainGenerator`?

**Comment by taruti** [comment]
I'll rebase to get the new conflict.

**Comment by taruti** [comment]
Resolved conflict with recent master changes.

**Comment by chrisnojima** [comment]
Not currently.  Generator is to be used rarely if possible

**Pull request merged**